### PR TITLE
[2.19.x] G-9824 Greatly speeds up search form GET requests by not creating a JAXBContext for each form being processed

### PR DESCRIPTION
**Pull request by Lambeaux**
#### What does this PR do?
Search form read operations for a single form take noticeably longer than other ops. There's a noticeable delay while debugging GET ops over lines that create `FilterReader` in `TemplateTransformer`. When this compounds for many search forms, the user will be left waiting. 

This PR fixes the issue, which is the creation of a `JAXBContext`, not per request, but **per form** being processed. This is causing redundant JAXB initialization processing of XML metadata that isn't necessary and now it only gets created once when the application starts: 
https://stackoverflow.com/a/7400735

This was missed the first time through because the javadocs made no mention of thread safety. 
https://docs.oracle.com/javase/8/docs/api/javax/xml/bind/JAXBContext.html

#### Ask 2 committers to review/merge the PR and tag them here.
@brianfelix
@cassandrabailey293
@millerw8
@mojogitoverhere

#### How should this be tested?
Create a bunch of search forms of various shapes and sizes, then refresh the page several times, repeatedly. Best case scenario would be multiple users requesting their forms at the same time. 

#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Threat Dragon models
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

File: catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/forms/filter/FilterReader.java [hunk 1]
```diff
@@ -50,6 +46,9 @@ public FilterReader() throws JAXBException {
   @SuppressWarnings("unchecked")
   private <T> JAXBElement<T> unmarshal(InputStream inputStream, Class<T> tClass)
       throws JAXBException {
+    SAXParserFactory factory = XMLUtils.getInstance().getSecureSAXParserFactory();
+    factory.setNamespaceAware(true);
```
Comment (Lambeaux) [inline catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/forms/filter/FilterReader.java:50]: It's not fair to assume the factory is thread safe, so this gets moved into the function and created on every `unmarshal` call.

File: catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/forms/TemplateTransformer.java [hunk 2]
```diff
@@ -56,13 +56,21 @@
 public class TemplateTransformer {
   private static final Logger LOGGER = LoggerFactory.getLogger(TemplateTransformer.class);
 
+  private final FilterReader reader;
```
Comment (Lambeaux) [inline catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/forms/TemplateTransformer.java:59]: Moving the factory around in the other file makes it safe to do this - and now the same JAXBContext gets used on every request.

**Comment by Lambeaux** [comment]
Build now.

**Review by vinamartin — APPROVED** [review]

**Review by cassandrabailey293 — APPROVED** [review]

**Review by brianfelix — APPROVED** [review]

**Review by mojogitoverhere — APPROVED** [review]

**Comment by cassandrabailey293** [comment]
hero in progress

**Comment by cassandrabailey293** [comment]
🎉 
Created 4-5 dsearch forms each for two different users. Verified that two users can refresh and get their search forms repeatedly in reasonable time. LGTM!

**Pull request merged**